lib/internal/crypto/hashnames.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const {
+  ObjectKeys
+} = primordials;
+
+const kHashContextNode = 1;
+const kHashContextWebCrypto = 2;
+const kHashContextJwkRsa = 3;
+const kHashContextJwkRsaPss = 4;
+const kHashContextJwkRsaOaep = 5;
+const kHashContextJwkHmac = 6;
+const kHashContextJwkDsa = 7;
+
+// WebCrypto and JWK use a bunch of different names for the
+// standard set of SHA-* digest algorithms... which is ... fun.
+// Here we provide a utility for mapping between them in order
+// make it easier in the code.
+
+const kHashNames = {
+  sha1: {
+    [kHashContextNode]: 'sha1',
+    [kHashContextWebCrypto]: 'SHA-1',
+    [kHashContextJwkRsa]: 'RS1',
+    [kHashContextJwkRsaPss]: 'PS1',
+    [kHashContextJwkRsaOaep]: 'RSA-OAEP',
+    [kHashContextJwkHmac]: 'HS1',
+    [kHashContextJwkDsa]: 'NODE-DSA-SHA-1',
+  },
+  sha256: {
+    [kHashContextNode]: 'sha256',
+    [kHashContextWebCrypto]: 'SHA-256',
+    [kHashContextJwkRsa]: 'RS256',
+    [kHashContextJwkRsaPss]: 'PS256',
+    [kHashContextJwkRsaOaep]: 'RSA-OAEP-256',
+    [kHashContextJwkHmac]: 'HS256',
+    [kHashContextJwkDsa]: 'NODE-DSA-SHA-256',
+  },
+  sha384: {
+    [kHashContextNode]: 'sha384',
+    [kHashContextWebCrypto]: 'SHA-384',
+    [kHashContextJwkRsa]: 'RS384',
+    [kHashContextJwkRsaPss]: 'PS384',
+    [kHashContextJwkRsaOaep]: 'RSA-OAEP-384',
+    [kHashContextJwkHmac]: 'HS384',
+    [kHashContextJwkDsa]: 'NODE-DSA-SHA-384',
+  },
+  sha512: {
+    [kHashContextNode]: 'sha512',
+    [kHashContextWebCrypto]: 'SHA-512',
+    [kHashContextJwkRsa]: 'RS512',
+    [kHashContextJwkRsaPss]: 'PS512',
+    [kHashContextJwkRsaOaep]: 'RSA-OAEP-512',
+    [kHashContextJwkHmac]: 'HS512',
+    [kHashContextJwkDsa]: 'NODE-DSA-SHA-512',
+  }
+};
+
+{
+  // Index the aliases
+  const keys = ObjectKeys(kHashNames);
+  for (let n = 0; n < keys.length; n++) {
+    const contexts = ObjectKeys(kHashNames[keys[n]]);
+    for (let i = 0; i < contexts.length; i++) {
+      const alias = kHashNames[keys[n]][contexts[i]];
+      if (kHashNames[alias] === undefined)
+        kHashNames[alias.toLowerCase()] = kHashNames[keys[n]];
+    }
+  }
```

Comment:
```suggestion
  const keys = ObjectKeys(kHashNames);
  for (const key of keys) {
    const contexts = ObjectKeys(kHashNames[key]);
    for (const context of contexts) {
      const alias = kHashNames[key][context].toLowerCase();
      if (kHashNames[alias] === undefined)
        kHashNames[alias] = kHashNames[key];
    }
  }
```

Reply:
I'm going to keep this one with the traditional for loop.

Reply:
Okay, but I also changed the place where toLowerCase is called because it seemed wrong.